src/handler.ts

```
@@ -43,13 +83,225 @@ function setupAWS() {
     const AWS_SDK_CONFIG = {
         credentials: {
             accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
         },
-        region: process.env.AWS_REGION
+        region: process.env.AWS_REGION,
     };
-    
+
     AWS.config.credentials = AWS_SDK_CONFIG.credentials;
     AWS.region = AWS_SDK_CONFIG.region;
-    
-    return new AWS.S3();
-}
\ No newline at end of file
+
+    if (process.env.MODE === 'dev') {
+        return new AWS.S3({ endpoint: process.env.LOCALSTACK_ENDPOINT_S3, s3ForcePathStyle: true });
+    } else {
+        return new AWS.S3({endpoint: process.env.DEV_ENDPOINT_S3, params: {Bucket: bucketName}, region: 'us-east-1'});
+    }
+}
+
+/**
+ * Copies files in s3 to location of new author
+ * @param token token for multipart file uploads
+ * @param fromCuid the cuid of the object to move
+ * @param oldAuthorAccessID the file access id of the old author
+ * @param newAuthorAccessID the file access id of the new author
+ */
+
+async function copyFiles(fromCuid, oldAuthorAccessID, newAuthorAccessID) {
+
+    const oldPrefix = `${oldAuthorAccessID.fileAccessId}/${fromCuid}`;
+    const newPrefix = `${newAuthorAccessID.fileAccessId}/${fromCuid}`;
+    try {
+        s3.listObjectsV2({Prefix: oldPrefix}, function (err, data) {
+            if (err) {
+                console.log(err, err.stack); // logs if an error occurs
```

Can we throw this error so it can be returned to the client?

---

all been thrown below